store/src/java/com/zimbra/cs/event/logger/EventLogger.java

```diff
@@ -154,8 +156,14 @@ public void shutdownEventNotifierExecutor() {
         }
 
         ZimbraLog.event.warn("Shutdown called for Event Notifier Executor; initiating shutdown sequence...");
-        executorService.shutdownNow();
+        shutdownExecutor.set(true);
+        executorService.shutdown();
         try {
+            if(drainQueueBeforeShutdown.get()) {
+                for (int i = 0; i < config.getNumThreads(); i++) {
+                    eventQueue.put(new Event("POISON_PILL_ACCOUNT_ID", Event.EventType.POISON_PILL, System.currentTimeMillis()));
```

Comment:
This is a personal preference, so feel free to disregard - would a singleton poison pill make sense here? That way you don't need a `POISON_PILL_ACCOUNT_ID` constant here... something like `eventQueue.put(Event.POISON_PILL)`?